sunpy/io/_fits.py

```diff
@@ -99,6 +100,15 @@ def read(filepath, hdus=None, memmap=None, **kwargs):
                 message += repr(e)
                 warn_user(message)
 
+            if memmap == True and not (isinstance(pairs[0].data.base, mmap.mmap) and pairs[0].data.base is not None):
+                try:
+                    if pairs[0].header['BZERO'] != 0:
```

Comment:
Is this actually a reliable way to detect a compressed image header?

Reply:
No, it isn't for two reasons.  See https://github.com/sunpy/sunpy/pull/6837#issuecomment-1479628183